main/src/library/Int16.flix

```diff
@@ -0,0 +1,170 @@
+     */
+    def size: Int32 = 16
+
+    /**
+     * Returns frequently used Int16 values for readability.
+     */
+    def zero: Int16 = 0i16
+    def one: Int16 = 1i16
+    def two: Int16 = 2i16
+
+    /**
+     * Returns the minimum number representable by an `Int16`.
+     */
+    def minValue: Int16 = one() << (size() - 1)
+
+    /**
+     * Returns the maximum number representable by an `Int16`.
+     */
+    def maxValue: Int16 = ~minValue()
+
+    /**
+     * Returns the smaller of `x` and `y`.
+     */
+    def min(x: Int16, y: Int16): Int16 = if (x <= y) x else y
+
+    /**
+     * Returns the larger of `x` and `y`.
+     */
+    def max(x: Int16, y: Int16): Int16 = if (x >= y) x else y
+
+    /**
+     * Returns the absolute value of `x`.
+     * If the absolute value exceeds maxValue(), -1 is returned
+     */
+    def abs(x: Int16): Int16 = switch {
+        case x >= zero()        => x
+        case x == minValue()    => -one()
+        case x != minValue()    => -x
+    }
+
+    /**
+     * Returns the distance between `x` and `y`.
+     * If this distance exceeds maxValue(), -1 is returned
+     */
+    def dist(x: Int16, y: Int16): Int16 = switch {
+        case x >= zero() && y >= zero()                 => abs(x - y)
+        case x < zero() && y < zero()                   => abs(x - y)
+        case x == minValue() || y == minValue()         => -one()
+        case minValue() + abs(x) + abs(y) >= zero()     => -one()
+        case minValue() + abs(x) + abs(y) < zero()      => abs(x - y)
+    }
+
+    /**
+     * Returns 1 if x > y, -1 if x < y, and 0 if x = y.
+     * The sign of x - y
+     */
+    def compare(x: Int16, y: Int16): Int32 = switch {
+        case x == y  => 0
+        case x < y   => -1
+        case x > y   => 1
+    }
+
+    /**
+     * Returns 1 if x > 0, -1 if x < 0, and 0 if x = 0.
+     * This sign of x
+     */
+    def signum(x: Int16): Int32 = compare(x, zero())
+
+    /**
+     * Returns the position of the highest-order/leftmost one-bit in `x`
+     * Possible return values: 0 (rightmost bit) - 15 (leftmost bit)
+     *                         -1 if x = 0
+     */
+    def highestOneBitPosition(x: Int16): Int32 =
+        // Start at bit 15 and scan right
+        oneBitPositionHelper(x, size() - 1, -1)
+
+    /**
+     * Returns the position of the lowest-order/rightmost one-bit in `x`
+     * Possible return values: 0 (rightmost bit) - 15 (leftmost bit)
+     *                         -1 if x = 0
+     */
+    def lowestOneBitPosition(x: Int16): Int32 =
+        // Start at bit 0 and scan left
+        oneBitPositionHelper(x, 0, 1)
+
+    /**
+     * Helper function for highestOneBitPosition and lowestOneBitPosition
+     * If `x` has a one-bit at `position`, return that position
+     * Otherwise recursively check the next bit in the same way
+     */
+    def oneBitPositionHelper(x: Int16, position: Int32, delta: Int32): Int32 = switch {
```

Comment:
Is this a "private" helper function or part of the public API?

Flix doesn't have `private`, `protected`, and `public`, and it probably never will. But we could consider (a) Writing "private" functions - i.e. functions we do not consider part of the public AP - with an underscore as in `_fooBar`, or (b) leave it out of the documentation.